digitaltwin/modelrepo/ModelServiceClient.cs

```diff
@@ -0,0 +1,397 @@
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoMapper;
+using Azure;
+using Azure.IoT.DigitalTwin.Model.Service.Generated;
+using Azure.IoT.DigitalTwin.Model.Service.Generated.Models;
+using Microsoft.Azure.Devices.Common.Authorization;
+using Microsoft.Rest;
+
+namespace Microsoft.Azure.DigitalTwin.Model.Service
+{
+    public class ModelServiceClient
+    {
+        private DigitalTwinRepositoryService digitalTwinRepositoryService;
+        public const string _apiVersion = "2019-07-01-Preview";
+        public string RepositoryId { get; set; }
+
+        /// <summary> Initializes a new instance of the <see cref="ModelServiceClient"/> class.</summary>
+        protected ModelServiceClient()
+        {
+
+        } 
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ModelServiceClient"/> class.</summary>
+        /// <param name="connectionString">Your Model Repo's connection string.</param>
+        public ModelServiceClient(string connectionString)
+        {
+            GuardHelper.ThrowIfNullOrWhiteSpace(connectionString, nameof(connectionString));
+
+            var modelConnectionStringParser = ModelServiceConnectionStringParser.CreateForModel(connectionString);
+            this.RepositoryId = modelConnectionStringParser.RespositoryId;
+            ModelServiceConnectionString modelServiceConnectionString = new ModelServiceConnectionString(modelConnectionStringParser);
+            IoTServiceClientCredentials serviceClientCredentials = new ModelSharedAccessKeyCredentials(modelServiceConnectionString);
+            this.SetupModelServiceClient(modelServiceConnectionString.HttpsEndpoint, serviceClientCredentials);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ModelServiceClient"/> class.
+        /// The client will use the provided endpoint and will generate credentials for each request using the provided <paramref name="credentials"/>.
+        /// </summary>
+        /// <param name="endpoint">The endpoint to connect to.</param>
+        /// <param name="credentials">The SAS token provider to use for authorization.</param>
+        /// <param name="repositoryId">Repository Id of private/company repository.</param>
+        public ModelServiceClient(Uri endpoint, IoTServiceClientCredentials credentials, string repositoryId)
+        {
+            GuardHelper.ThrowIfNull(endpoint, nameof(endpoint));
+            GuardHelper.ThrowIfNull(credentials, nameof(credentials));
+            GuardHelper.ThrowIfNull(repositoryId, nameof(repositoryId));
+            this.RepositoryId = repositoryId;
+            this.SetupModelServiceClient(endpoint, credentials);
+        }
+
+        private void SetupModelServiceClient(Uri endpoint, IoTServiceClientCredentials credentials)
+        {
+            DelegatingHandler[] handlers = new DelegatingHandler[1] { new AuthorizationDelegatingHandler(credentials) };
```

Comment:
The number in the array is unnecessary, because of the way the array is initialized. It will auto-size.